### PR TITLE
Cleanup redundant code for computing t complexity

### PR DESCRIPTION
**Pull request by anurudhp**
Cleaning up some redundant code to compute t-complexity.

- remove `_get_all_rotation_types` (in favor of `bloq_is_rotation`).
- reduce uses of `t_counts_from_sigma` (in favor of `QECGatesCost`).

TODOs in follow up:
- Currently, only `t_counts_from_sigma` uses eps correctly, but it's only used in a handful of places. This should be replaced with `QECGatesCost` after adding support for eps (#1250).
- We could add a simple `GateCounts.total_t_and_rotation_count` helper to support existing use cases which only take number of rotations (e.g. Beverland-et-al model).

Relevant discussions: 
- https://github.com/quantumlib/Qualtran/issues/662#issuecomment-1949215534
- https://github.com/quantumlib/Qualtran/pull/678#discussion_r1492944645

### CODE REVIEW

File: qualtran/resource_counting/t_counts_from_sigma_test.py [hunk 1]
```diff
@@ -11,66 +11,54 @@
...
-    }
-    expected_t_count = (
-        +100
-        + 1 * TComplexity.rotation_cost(z_eps1)
-        + 5 * TComplexity.rotation_cost(z_eps2)
-        + 9 * TComplexity.rotation_cost(x_eps)
-        + 13 * TComplexity.rotation_cost(y_eps)
-        + 20 * TComplexity.rotation_cost(cz_eps)
-    )
-    assert t_counts_from_sigma(sigma) == expected_t_count
```
Comment (mpharrigan) [inline qualtran/resource_counting/t_counts_from_sigma_test.py:76]: can you revert this test. This actually tests the expected behavior of `t_counts_from_sigma`, i.e. it has a sigma with more than one entry

**Review by mpharrigan — COMMENTED** [review]
Generally very reasonable changes; unsure about that one test deletion

File: qualtran/resource_counting/t_counts_from_sigma_test.py (hunk 1 above)
Comment (anurudhp) [reply qualtran/resource_counting/t_counts_from_sigma_test.py]: ah makes sense, reverted.

File: qualtran/resource_counting/t_counts_from_sigma.py [hunk 2]
```diff
@@ -11,45 +11,26 @@
...
+    from qualtran.resource_counting.classify_bloqs import bloq_is_rotation
 
-    if rotation_types is None:
-        rotation_types = _get_all_rotation_types()
     ret = sigma.get(TGate(), 0) + sigma.get(TGate().adjoint(), 0)
     for bloq, counts in sigma.items():
-        if isinstance(bloq, rotation_types) and not cirq.has_stabilizer_effect(bloq):
+        if bloq_is_rotation(bloq) and not cirq.has_stabilizer_effect(bloq):
+            if isinstance(bloq, Controlled):
+                # TODO native controlled rotation bloqs missing (CRz, CRy etc.)
```
Comment (mpharrigan) [inline qualtran/resource_counting/t_counts_from_sigma.py:32]: can you link that issue #878

**Review by mpharrigan — APPROVED** [review]

**Pull request merged**